pre_commit_hooks/nbsmoke_lint.py

```diff
@@ -11,19 +14,29 @@ def main(argv=None):
     parser.add_argument("filenames", nargs="*", help="Filenames to check.")
     args = parser.parse_args(argv)
 
-    return pytest.main(
-        [
-            # Avoid printing nbsmoke warnings.
-            # Most seem irrelevant for PySpark notebooks.
-            "-Wignore",
-            # Ignore undefined 'spark' for PySpark notebooks.
-            "-o",
-            "nbsmoke_flakes_to_ignore=undefined name 'spark'",
-            # Perform nbsmoke lint check.
-            "--nbsmoke-lint",
-        ]
-        + args.filenames
-    )
+    # Copy all notebooks to a temporary directory for the lint check.
+    # Without copying, pytest will try to import the notebook folders as python modules
+    # which is unnecessary and may fail if we don't have the required dependencies.
+    with tempfile.TemporaryDirectory() as tmp_dir:
```

Comment:
Why does copying to a new directory change the behavior?

Reply:
Yeah, copying to a new folder means `__init__.py` won't exist, so the file won't get run when pytest imports the folder as a python module